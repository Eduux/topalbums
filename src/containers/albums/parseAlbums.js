export default album => ({
  id: album.id.attributes['im:id'],
  image: album['im:image'][2].label,
  name: album['im:name'].label,
  tracksCount: album['im:itemCount'].label,
  artist: album['im:artist'].label,
  category: album.category.attributes.term,
});
