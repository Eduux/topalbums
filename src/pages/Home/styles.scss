@import '~/styles/theme.scss';

.homeContainer {
  @include container;
  display: flex;
  flex-direction: column;
}
