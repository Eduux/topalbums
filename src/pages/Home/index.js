/* eslint-disable no-console */
import React, { useEffect } from 'react';

import { useAlbums } from '~/containers/albums';

import Banner from '~/components/Banner';
import ListAlbums from '~/components/ListAlbums';

import './styles.scss';

const Home = () => {
  const {
    data: { list, loading, error },
    actions: { listAlbums },
  } = useAlbums();

  useEffect(() => {
    if (!list.length) listAlbums();
    // eslint-disable-next-line
  }, []);

  return error ? (
    <div>erro</div>
  ) : (
    <div>
      {loading ? (
        <div>carregando</div>
      ) : (
        <div className="homeContainer">
          <Banner />
          <ListAlbums list={list} />
        </div>
      )}
    </div>
  );
};

export default Home;
