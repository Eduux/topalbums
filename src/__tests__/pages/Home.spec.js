import React from 'react';
import { render } from '@testing-library/react';

import Home from '~/pages/Home';

import parseAlbums from '~/containers/albums/parseAlbums';

import albumRequest from '~/__mocks__/requests/albumsRequest';

import withAlbums, { initialState } from '~/containers/albums';

describe('Home page', () => {
  it('should render without data', () => {
    const Container = withAlbums(Home, initialState);

    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });

  it('must render with error', () => {
    const Container = withAlbums(Home, {
      ...initialState,
      error: 'An unexpected error has occurred!',
    });

    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });

  it('must render with data', () => {
    const Container = withAlbums(Home, {
      ...initialState,
      list: [albumRequest.feed.entry.map(parseAlbums)],
    });

    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });

  it('must render with loading', () => {
    const Container = withAlbums(Home, {
      ...initialState,
      loading: true,
    });

    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });
});
