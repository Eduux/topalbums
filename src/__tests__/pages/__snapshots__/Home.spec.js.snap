// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page must render with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="homeContainer"
        >
          <div
            class="bannerContainer"
          >
            <h2>
              Top 100 Albums Itunes
            </h2>
          </div>
          <div
            class="listAlbumsContainer"
          >
            <div
              class="listItemContainer"
            >
              <div
                class="positionAlbumContainer"
              >
                #1
              </div>
              <img
                alt="Album image 0"
                height="auto"
                loading="lazy"
                width="120"
              />
              <p
                class="listItemAlbumName"
              />
              <p
                class="listItemAlbumTracks"
              >
                undefined tracks
              </p>
              <div
                class="listItemAlbumArtist"
              >
                <div>
                  <svg
                    fill="currentColor"
                    height="40"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="40"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12,4c-4.411,0-8,3.589-8,8s3.589,8,8,8s8-3.589,8-8S16.411,4,12,4z M8,12H6c0-3.309,2.691-6,6-6v2 C9.794,8,8,9.794,8,12z M11.993,14c-1.108,0-2.007-0.898-2.007-2.007s0.898-2.007,2.007-2.007S14,10.885,14,11.993 S13.102,14,11.993,14z"
                      fill="none"
                    />
                    <circle
                      cx="11.993"
                      cy="11.993"
                      r="2.007"
                    />
                    <path
                      d="M12,2C6.486,2,2,6.486,2,12s4.486,10,10,10s10-4.486,10-10S17.514,2,12,2z M12,20c-4.411,0-8-3.589-8-8s3.589-8,8-8 s8,3.589,8,8S16.411,20,12,20z"
                    />
                    <path
                      d="M12,6c-3.309,0-6,2.691-6,6h2c0-2.206,1.794-4,4-4V6z"
                    />
                  </svg>
                </div>
                <div>
                  <p />
                  <p />
                </div>
                <div>
                  <svg
                    data-testid="favorite-icon"
                    fill="currentColor"
                    height="25"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="homeContainer"
      >
        <div
          class="bannerContainer"
        >
          <h2>
            Top 100 Albums Itunes
          </h2>
        </div>
        <div
          class="listAlbumsContainer"
        >
          <div
            class="listItemContainer"
          >
            <div
              class="positionAlbumContainer"
            >
              #1
            </div>
            <img
              alt="Album image 0"
              height="auto"
              loading="lazy"
              width="120"
            />
            <p
              class="listItemAlbumName"
            />
            <p
              class="listItemAlbumTracks"
            >
              undefined tracks
            </p>
            <div
              class="listItemAlbumArtist"
            >
              <div>
                <svg
                  fill="currentColor"
                  height="40"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="40"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12,4c-4.411,0-8,3.589-8,8s3.589,8,8,8s8-3.589,8-8S16.411,4,12,4z M8,12H6c0-3.309,2.691-6,6-6v2 C9.794,8,8,9.794,8,12z M11.993,14c-1.108,0-2.007-0.898-2.007-2.007s0.898-2.007,2.007-2.007S14,10.885,14,11.993 S13.102,14,11.993,14z"
                    fill="none"
                  />
                  <circle
                    cx="11.993"
                    cy="11.993"
                    r="2.007"
                  />
                  <path
                    d="M12,2C6.486,2,2,6.486,2,12s4.486,10,10,10s10-4.486,10-10S17.514,2,12,2z M12,20c-4.411,0-8-3.589-8-8s3.589-8,8-8 s8,3.589,8,8S16.411,20,12,20z"
                  />
                  <path
                    d="M12,6c-3.309,0-6,2.691-6,6h2c0-2.206,1.794-4,4-4V6z"
                  />
                </svg>
              </div>
              <div>
                <p />
                <p />
              </div>
              <div>
                <svg
                  data-testid="favorite-icon"
                  fill="currentColor"
                  height="25"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="25"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home page must render with error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        erro
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      erro
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home page must render with loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          carregando
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        carregando
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home page should render without data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          carregando
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        carregando
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
